os/booting-on-ecs.md

```diff
@@ -21,21 +21,37 @@ coreos:
      runtime: true
      content: |
        [Unit]
-       Description=Amazon ECS Agent
-       After=docker.service
-       Requires=docker.service
-       Requires=network-online.target
-       After=network-online.target
-       
+       Description=AWS ECS Agent (https://github.com/aws/amazon-ecs-agent)
+       Documentation=https://docs.aws.amazon.com/AmazonECS/latest/developerguide/
+       Requires=network-online.target docker.socket
```

Comment:
Well, my only concern is that "network-online.target" is kind of a hack and is a little racy. The better solution is to create a service which is tolerant of a network failure and can retry. In this case, the service is already tolerant to these failures (if docker fails to pull the image, the service will restart after 30 seconds and the ECS agent itself should be tolerant to these failures as well). I think it is safe to omit the dependency entirely.

The other thing to consider is that the Docker service itself should declare its dependency on the network. Because you are then depending on Docker, the dependency on the network is implicit.